### PR TITLE
Added Notes on Auth* to README

### PR DESCRIPTION
**Pull request by clintonb**
New notes instruct developers how to get work around auth system. auth_auth view now redirects to demo course page.

@mulby @dsjen @rocha

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,6 +41,16 @@ Need a fallback to disable a feature? Create a [Waffle](http://waffle.readthedoc
 See the [Waffle documentation](http://waffle.readthedocs.org/en/latest/) for details on utilizing features in code and templates.
 
 
+Authentication & Authorization
+------------------------------
+By default, this application relies on an external OAuth2/Open ID Connect provider 
+(contained within the [LMS](https://github.com/edx/edx-platform)) for authentication and authorization. If you are a 
+developer, and do not want to setup edx-platform, you can get around this requirement by doing the following:
+
+1. Set `ENABLE_AUTO_AUTH` to `True` in your settings file. (This is the default value in `settings/local.py`).
```
Comment (rocha) [inline README.md:50]: I am not a 100% sure about the name. What about `DISABLE_AUTHENTICATION`? Does that make sense?

**Comment by rocha** [comment]
Looks good, just one observation about the setting name.

File: README.md (hunk 1 above)
Comment (clintonb) [inline README.md:50]: The setting doesn't disable authentication entirely. It enables a special view that creates and logs in a new user. How about `ENABLE_DEV_VIEWS` or something along those lines?

Comment (rocha) [inline README.md:50]: `DEBUG_AUTH_VIEWS`? After your clarification, `ENABLE_AUTO_AUTH` doesn't sound that bad, maybe add an extra comment.

**Pull request merged**